### ISSUE TITLE
Double click in Unity3D on a script should open Consulo on that script

### ISSUE THREAD

**Issue by netwerkz** [consulo/consulo-unity3d#1]
I guess read and parse the arguments passed by Unity3D to Consulo and use them to bring to front or open same script that was double clicked from Unity. Minor convenience feature :) but a must have for the Unity plugin

**Comment by VISTALL** [comment]
Main problem - that Unity3D cant create Consulo project files. Project files are required

**Comment by netwerkz** [comment]
Are you sure you can't read the class name or filepath from the arguments passed from Unity?

**Comment by VISTALL** [comment]
Unity have option for use External Tool for editing, maybe Consulo can open file, but without project its a problem

**Comment by ghost** [comment]
![image](https://cloud.githubusercontent.com/assets/10781622/5991095/7f4012a0-aa11-11e4-97ae-16a6fbb74d35.png)
when open a cs file in unity

**Comment by VISTALL** [comment]
Dont try set Consulo as external tools) Consulo cant editor file without opened project

**Comment by VISTALL** [comment]
Its not possible to add New App to this list(it is not enough only command line). MonoDevelop + UnityVS hardcoded inside UnityEditor :(

**Comment by commovere** [comment]
So maybe we should create some issue in unity tracker, so community will vote for the same for consulo

**Comment by VISTALL** [comment]
I already saw this - http://feedback.unity3d.com/suggestions/get-rid-of-monodevelop-legacy-and-build-a-custom-ide-based-on-intellij-idea  

Very small votes count. I think no sence to create new ticket (for now maybe)

**Comment by ernestp** [comment]
Seems that unity hear us:

> SchubkraftJul 17, 2015 - We are not going to build our own version of anything but started on implementing an easier way to use external IDEs with Unity. See the http://unity3d.com/unity/roadmap

But seems 99% of time we using existing project, not creating new projects. It not necessary should be Consulo app itself. Let it be external tool that check if project is created, show warning if not and open file in Consulo and go to specified line if project is already created.

Honestly seems that is last rubicon to have comfort work in pair Unity - Consulo

**Comment by VISTALL** [comment]
They show their roadmap https://unity3d.com/unity/roadmap

I am waiting **MonoDevelop and Unity REST Integration** at 5.4 MAR 16, 2016

**Comment by minhdu** [comment]
Hi everyone, i was write a small plugin for Unity that support open Consulo and focus exactly file/line by double click on script or debug console:
https://github.com/minhdu/UnityConsulo

**Comment by VISTALL** [comment]
Hi guys. I create some library (thanks @minhdu) with some my modifications.

Please read README from https://github.com/consulo/UnityEditorConsuloPlugin how to use.

It required lastest Consulo & Unity plugin builds

Thanks

**Comment by hjoek** [comment]
Still having problems with this...

I first just tried to set Consulo as the external script editor in Unity's settings. This resulted in Consulo being focused on double click but the right file or line did not open.

Then I followed the instructions in  https://github.com/consulo/UnityEditorConsuloPlugin, installed the plugin and checked "Edit/Use Consulo as External Editor". And behold: Now the correct file and correct line (or close enough) is opened in Consulo when I double click something in Unity! But alas: Now Consulo app is NOT focused!! I need to focus it manually after double clicking something.

Could it be possible to get it working so, that when I double click something in Unity, the corresponding line would be opened in Consulo AND the Consule app would be focused at the same time?

I'm using Mac with Unity 3d v5.3.4f1 and Consulo 1.0 #1318.
